tools/language-server/logstat/src/main.rs

```diff
@@ -0,0 +1,317 @@
+            }
+            op.duration = op.timestamp - timestamp;
+            timestamp = op.timestamp;
+        }
+    }
+}
+
+/// Analyze benchmark results
+fn analyze_iterations(iterations: &[Iteration], use_median: bool) -> Vec<Stats> {
+    let mut stats = vec![];
+
+    let ops_len = iterations.len();
+    let splits_len = iterations.first().unwrap().operations.len();
+
+    let mut i = 0;
+    while i < splits_len {
+        let mut min = Duration::MAX;
+        let mut max = Duration::ZERO;
+        let mut sum = Duration::ZERO;
+        let mut durations = vec![];
+
+        let current_line = &iterations[0].operations[i].line;
+
+        for iteration in iterations {
+            let timed = &iteration.operations[i];
+
+            durations.push(timed.duration);
+            min = timed.duration.min(min);
+            max = timed.duration.max(max);
+            sum += timed.duration;
+        }
+
+        let avg = if use_median { median(&mut durations) } else { sum / ops_len as u32 };
+        let line = current_line.to_string();
+
+        stats.push(Stats { min, max, avg, line });
+
+        i += 1;
+    }
+
+    let overall_stats = iterations_average(iterations, &stats);
+    stats.push(overall_stats);
+
+    stats
+}
+
+/// Calculate the average of all iterations
+fn iterations_average(ops: &[Iteration], stats: &[Stats]) -> Stats {
+    let min_opt = ops.into_iter().map(|o| o.total_time()).min();
+    let max_opt = ops.into_iter().map(|o| o.total_time()).max();
+
+    let min = min_opt.unwrap_or(Duration::ZERO);
+    let max = max_opt.unwrap_or(Duration::ZERO);
+    let avg = stats.into_iter().map(|s| s.avg).sum();
+    let line = String::from("Total");
+
+    Stats { min, max, avg, line }
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let args = Args::parse();
+    let spec = read_specs(&args.spec).await?;
+
+    let mut log_iterations = read_logfile(&args.log, &spec).await?;
+
+    let mut iterations;
+    let mut ws_iterations;
+
+    if let Some(path_buf) = args.wstest_log {
+        ws_iterations = read_logfile(&path_buf, &WSTEST_SPEC).await?;
+
+        // skip warmup iterations
+        let start_time = &ws_iterations[0].operations[0].timestamp;
+        log_iterations = log_iterations
+            .into_iter()
+            .skip_while(|iteration| {
+                let first_operation = &iteration.operations[0];
+                &first_operation.timestamp < start_time
+            })
+            .collect();
+
+        if ws_iterations.len() != log_iterations.len() {
+            eprintln!(
+                "[ERR] Unequal number of benchmark iterations in log files! [{}] vs. [{}]",
+                ws_iterations.len(),
+                log_iterations.len()
+            );
+            process::exit(1);
+        }
+
+        merge_iterations(&mut ws_iterations, log_iterations);
+
+        iterations = ws_iterations;
+    } else {
+        iterations = log_iterations;
+    }
```

Comment:
You can avoid mutable variables (which is always cool in terms of core readability) this way:
```rust
    let mut iterations;
    let mut ws_iterations;

    let iterations = if let Some(path_buf) = args.wstest_log {
        // blah blah blag...
        ws_iterations
    } else {
        log_iterations
    };
```